src/panels/lovelace/editor/hui-badge-preview.ts

```diff
@@ -0,0 +1,99 @@
+import { HomeAssistant } from "../../../types";
+import { LovelaceBadgeConfig } from "../../../data/lovelace";
+import { ConfigError } from "./types";
+import { computeRTL } from "../../../common/util/compute_rtl";
+import { LovelaceBadge } from "../types";
+import { createBadgeElement } from "../create-element/create-badge-element";
+import { createErrorBadgeConfig } from "../badges/hui-error-badge";
+
+import "../../../components/entity/ha-state-label-badge";
+
+export class HuiBadgePreview extends HTMLElement {
+  private _hass?: HomeAssistant;
+  private _element?: LovelaceBadge;
+  private _config?: LovelaceBadgeConfig;
+
+  private get _error() {
+    return this._element?.tagName === "HUI-ERROR-CARD";
+  }
+
+  constructor() {
+    super();
+    this.addEventListener("ll-rebuild", () => {
+      this._cleanup();
+      if (this._config) {
+        this.config = this._config;
+      }
+    });
+  }
+
+  set hass(hass: HomeAssistant) {
+    if (!this._hass || this._hass.language !== hass.language) {
+      this.style.direction = computeRTL(hass) ? "rtl" : "ltr";
+    }
+
+    this._hass = hass;
+    if (this._element) {
+      this._element.hass = hass;
+    }
+  }
+
+  set error(error: ConfigError) {
+    this._createBadge(
+      createErrorBadgeConfig(`${error.type}: ${error.message}`)
+    );
+  }
+
+  set config(configValue: LovelaceBadgeConfig) {
+    const curConfig = this._config;
+    this._config = configValue;
+
+    if (!configValue) {
+      this._cleanup();
+      return;
+    }
+
+    if (!this._element) {
+      this._createBadge(configValue);
+      return;
+    }
+
+    // in case the element was an error element we always want to recreate it
+    if (!this._error && curConfig && configValue.type === curConfig.type) {
+      try {
+        this._element.setConfig(configValue);
+      } catch (err) {
+        this._createBadge(createErrorBadgeConfig(err.message));
```

Comment:
So we have 2 error elements, an `error-badge` we create here, and an `error-card` we create when the type is unknown in `create-element-base`.

The `error-badge` would only be used here, we should align that with the behavior of `hui-view` so either make that create an `error-badge` instead of `error-card` or create an `error-card` here.

Reply:
Will this even ever catch the error? As we catch it here: <https://github.com/home-assistant/frontend/blob/dev/src/panels/lovelace/create-element/create-element-base.ts#L62-L66>

Reply:
No it doesnt ever seem to hit it :)